cloudshell/cli/session/expect_session.py

```diff
@@ -134,83 +114,110 @@ def _clear_buffer(self, timeout):
+        :param command: command to send
+        :param expected_string: expected string
         :param expect_map: dict with {re_str: action} to trigger some action on received string
         :param error_map: expected error list
         :param timeout: session timeout
         :param retries: maximal retries count
         :return:
         """
+        if not command:
+            command = ''
 
         retries = retries or self._max_loop_retries
         empty_loop_timeout = empty_loop_timeout or self._empty_loop_timeout
 
-        if data_str is not None:
-            self._clear_buffer(self._clear_buffer_timeout)
+        if command is not None:
+            self._clear_buffer(self._clear_buffer_timeout, logger)
 
-            self.logger.debug('Command: {}'.format(data_str))
-            self.send_line(data_str)
+            logger.debug('Command: {}'.format(command))
+            self.send_line(command, logger)
 
-        if re_string is None or len(re_string) == 0:
-            raise Exception('ExpectSession', 'List of expected messages can\'t be empty!')
+        if not expected_string:
+            raise ExpectedSessionException(self.__class__.__name__, 'List of expected messages can\'t be empty!')
 
         # Loop until one of the expressions is matched or MAX_RETRIES
         # nothing is expected (usually used for exit)
         output_list = list()
         output_str = ''
         retries_count = 0
         is_correct_exit = False
+        command_removed = False
         action_loop_detector = ActionLoopDetector(self._loop_detector_max_action_loops,
                                                   self._loop_detector_max_combination_length)
 
         while retries == 0 or retries_count < retries:
 
             try:
-                read_buffer = self._receive(timeout)
+                # read_buffer = self._receive(timeout, logger)
+                read_buffer = self._receive_all(timeout, logger)
             except socket.timeout:
                 read_buffer = None
 
             if read_buffer:
+                read_buffer = normalize_buffer(read_buffer)
+                logger.info(read_buffer)
                 output_str += read_buffer
+                if command and not command_removed:
+                    command_pattern = '^.*' + command + '.*\\n'
+                    if re.search(command_pattern, output_str):
+                        output_str = re.sub(command_pattern, '', output_str)
+                        command_removed = True
                 retries_count = 0
             else:
                 retries_count += 1
                 time.sleep(empty_loop_timeout)
                 continue
 
-            if re.search(re_string, output_str, re.DOTALL):
+            if re.search(expected_string, output_str, re.DOTALL):
+                # logger.debug('Expected str: {}'.format(expected_string))
                 output_list.append(output_str)
                 is_correct_exit = True
```

Comment:
is_correct_exit - is not clear enough what is the purpose of the flag. Maybe we can call it expected_output_found
And add the description that this flag used to verify if we got the expected output from the switch. If the expected output found we exit the loop of retries.